src/sfdo.c

```diff
@@ -210,11 +211,19 @@ process_abs_name(struct icon_ctx *ctx, const char *icon_name)
 static struct sfdo_desktop_entry *
 get_db_entry_by_id_fuzzy(struct sfdo_desktop_db *db, const char *app_id)
 {
+	struct sfdo_desktop_entry *entry;
+
+	/* try direct match first */
+	entry = sfdo_desktop_db_get_entry_by_id(db, app_id, SFDO_NT);
```

Comment:
This is still OK for icon look up?

Reply:
Its doing duplicated work in case of icon lookup, suggest something like a generic `get_desktop_entry()` (or whatever its named) function within the loader which does all the weird matching (app_id, lowercase(app_id), wm_class match, partial match, ..). That function can then be used by both, the name and the icon lookup.

Reply:
Thx, I had indeed missed that this function is already used for icon look up.